packages/marshal/src/passStyleOf.js

```diff
@@ -235,50 +286,99 @@ const assertRemotableProto = val => {
  *
  * @param {*} val The remotable candidate to check
+ * @param {Checker} [check]
+ * @returns {boolean}
  */
-export function assertCanBeRemotable(val) {
-  // throws exception if cannot
-  assert.typeof(val, 'object', X`cannot serialize non-objects like ${val}`);
-  assert(!Array.isArray(val), X`Arrays cannot be pass-by-remote`);
-  assert(val !== null, X`null cannot be pass-by-remote`);
+function checkCanBeRemotable(val, check = x => x) {
+  if (
+    !(
+      check(
+        typeof val === 'object',
+        X`cannot serialize non-objects like ${val}`,
+      ) &&
+      check(!Array.isArray(val), X`Arrays cannot be pass-by-remote`) &&
+      check(val !== null, X`null cannot be pass-by-remote`)
+    )
+  ) {
+    return false;
+  }
 
   const descs = getOwnPropertyDescriptors(val);
   const keys = ownKeys(descs); // enumerable-and-not, string-or-Symbol
-  keys.forEach(key => {
-    assert(
+  return keys.every(
+    key =>
       // Typecast needed due to https://github.com/microsoft/TypeScript/issues/1863
-      !('get' in descs[/** @type {string} */ (key)]),
-      X`cannot serialize objects with getters like ${q(String(key))} in ${val}`,
-    );
-    assert.typeof(
-      // @ts-ignore https://github.com/microsoft/TypeScript/issues/1863
-      val[key],
-      'function',
-      X`cannot serialize objects with non-methods like ${q(
-        String(key),
-      )} in ${val}`,
-    );
-    assert(
-      key !== PASS_STYLE,
-      X`A pass-by-remote cannot shadow ${q(PASS_STYLE)}`,
-    );
-  });
+      check(
+        !('get' in descs[/** @type {string} */ (key)]),
+        X`cannot serialize objects with getters like ${q(
+          String(key),
+        )} in ${val}`,
+      ) &&
+      check(
+        typeof val[key] === 'function',
+        X`cannot serialize objects with non-methods like ${q(
+          String(key),
+        )} in ${val}`,
+      ) &&
+      check(
+        key !== PASS_STYLE,
+        X`A pass-by-remote cannot shadow ${q(PASS_STYLE)}`,
+      ),
+  );
 }
 
+const canBeRemotable = val => checkCanBeRemotable(val);
+harden(canBeRemotable);
+export { canBeRemotable };
+
+const assertCanBeRemotable = val => {
+  checkCanBeRemotable(val, assertChecker);
+};
+harden(assertCanBeRemotable);
+export { assertCanBeRemotable };
+
 /**
  * @param {Remotable} val
+ * @param {Checker} [check]
+ * @returns {boolean}
  */
-function assertRemotable(val) {
-  assert(isFrozen(val), X`cannot serialize non-frozen objects like ${val}`);
-
-  assertCanBeRemotable(val);
-
+function checkRemotable(val, check = x => x) {
+  const not = (cond, details) => !check(cond, details);
+  if (not(isFrozen(val), X`cannot serialize non-frozen objects like ${val}`)) {
+    return false;
+  }
+  if (!checkCanBeRemotable(val, check)) {
+    return false;
+  }
   const p = getPrototypeOf(val);
-  if (p !== null && p !== objectPrototype) {
-    assertRemotableProto(p);
+  if (p === null || p === objectPrototype) {
+    return true;
   }
```

Comment:
Put a TODO comment here to remove this once we insist that remotables be explicitly marked remotable.